package p2vera;

enum msg_types {
	link = 0; //Проверка связи. Тело сообщения позволяет проверить наличие связи и наличие прошлых связей между устройствами 
	info = 1; //Запрос информации об удаленном приложении. Позволяет получить его имя, идентификатор, загрузку и т.п.
}


message msg_wrapper {
	required int32 msg_type = 1; //Тип сообщения. На его основании выполняется обработка поля body
	required int32 msg_id   = 2; //Уникальный идентификатор этого сообщения. В сообщении запросе и сообщении ответа эти значения должны совпадать
	enum msg_direction {
		request  = 0;
		response = 1;
	}
	optional msg_direction dir  = 3 [default = request]; //Направление распространения сообщений. Позволяет отделить ответы от запросов
	optional bytes route     = 4; //Правила маршрутизации сообщения. Инкапсулируют структуры, определяющие способ доставки
	optional bytes body      = 5; //Содержимое сообщения. Инкапсулирует сообщения любых других типов
}



